src/main/kotlin/org/jetbrains/research/testspark/helpers/psiHelpers/JavaPsiHelper.kt

```diff
@@ -173,145 +136,313 @@ class JavaPsiHelper : PsiHelperInterface {
+    override fun generateMethodDescriptor(
+        psiMethod: PsiMethodWrapper,
+    ): String {
+        val methodDescriptor = psiMethod.methodDescriptor
+        log.info("Method description: $methodDescriptor")
+        return methodDescriptor
+    }
 
-        return (selectedLine in firstStatementLine..lastStatementLine)
+    override fun getSurroundingClass(
+        caretOffset: Int,
+    ): PsiClassWrapper? {
+        val classElements = PsiTreeUtil.findChildrenOfAnyType(psiFile, PsiClass::class.java)
+        for (cls in classElements) {
+            if (withinElement(cls, caretOffset)) {
+                val javaClassWrapper = JavaPsiClassWrapper(cls)
+                if (javaClassWrapper.isValid()) {
+                    log.info("Surrounding class for caret in $caretOffset is ${javaClassWrapper.qualifiedName}")
+                    return javaClassWrapper
+                }
+            }
+        }
+        log.info("No surrounding class for caret in $caretOffset")
+        return null
     }
 
-    /**
-     * Determines whether a given PsiElement contains a specified caret offset.
-     *
-     * @param psiElement The PsiElement to check if it contains the caret offset.
-     * @param caretOffset The caret offset to check.
-     * @return `true` if the PsiElement contains the caret offset, `false` otherwise.
-     */
-    private fun withinElement(
-        psiElement: PsiElement,
+    override fun getSurroundingMethod(
         caretOffset: Int,
-    ): Boolean {
-        return (psiElement.startOffset <= caretOffset) && (psiElement.endOffset >= caretOffset)
+    ): PsiMethodWrapper? {
+        val methodElements = PsiTreeUtil.findChildrenOfAnyType(psiFile, PsiMethod::class.java)
+        for (method in methodElements) {
+            if (method.body != null && withinElement(method, caretOffset)) {
+                val surroundingClass =
+                    PsiTreeUtil.getParentOfType(method, PsiClass::class.java) ?: continue
+                val surroundingClassWrapper = JavaPsiClassWrapper(surroundingClass)
+                if (surroundingClassWrapper.isValid()) {
+                    val javaMethod = JavaPsiMethodWrapper(method)
+                    log.info("Surrounding method for caret in $caretOffset is ${javaMethod.methodDescriptor}")
+                    return javaMethod
+                }
+            }
+        }
+        log.info("No surrounding method for caret in $caretOffset")
+        return null
     }
 
-    /**
-     * Gets the display name of a class, depending on if it is a normal class, an abstract class or an interface.
-     * This is used when displaying the name of a class in GenerateTestsActionClass menu entry.
-     *
-     * @param psiClass the PSI class of interest
-     * @return the display name of the PSI class
-     */
-    private fun getClassDisplayName(psiClass: PsiClass): String {
-        return if (psiClass.isInterface) {
+    override fun getSurroundingLine(
+        caretOffset: Int,
```

Comment:
It is unclear whether it returns a line index or a line number; could you mention it either in the Javadoc or method name?

Reply:
It is already stated in doc^ 
> The line number of the selected line, otherwise null.